coderd/promoauth/github.go

```diff
@@ -0,0 +1,92 @@
+package promoauth
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+type rateLimits struct {
+	Limit     int
+	Remaining int
+	Used      int
+	Reset     time.Time
+	Resource  string
+}
+
+// githubRateLimits checks the returned response headers and
+func githubRateLimits(resp *http.Response, err error) (rateLimits, bool) {
+	if err != nil || resp == nil {
+		return rateLimits{}, false
+	}
+
+	p := headerParser{header: resp.Header}
+	// See
+	// https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#checking-the-status-of-your-rate-limit
+	limits := rateLimits{
+		Limit:     p.int("x-ratelimit-limit"),
+		Remaining: p.int("x-ratelimit-remaining"),
+		Used:      p.int("x-ratelimit-used"),
+		Resource:  p.header.Get("x-ratelimit-resource"),
```

Comment:
For `int` we check if the key is missing and log an error, but here we accept empty string which could result in "", "-unauthorized", etc. Should we register an error in this case too?

Reply:
Yea, I was being a bit lazy. If the int values are missing, then we will always return an error. The `x-ratelimit-resource` will never be empty if the ints are set.

I can add a quick checker though and just make it consistent.